docs/references/calfits_memo.tex

```diff
@@ -1,167 +1,173 @@
+% We are defining a file format with the code interface. 
+We describe the required and optional parameters of a UVCal FITS---hereafter \textit{calfits}---file and pyuvdata's interface for reading and writing these files. 
+For usage examples please refer to the pyuvdata tutorial: \url{http://pyuvdata.readthedocs.io/en/latest/tutorial.html}.
+
+\section{Overview}
+\textit{Calfits} is an adaptation of the FITS file format to enable the storage of calibration information for radio interferometric arrays.
+Because it builds on the existing FITS file format, all \emph{calfits} files should be properly formatted according to the FITS standard, which is widely available\footnote{\url{https://fits.gsfc.nasa.gov/fits_documentation.html}}.
+
+
+Any valid \textit{calfits} file corresponds directly to a UVCal object within pyuvdata.
+As such, every new HDU keyword introduced here has a one-to-one correspondence with UVCal object parameters.
+In this memo, each new keyword is followed by its corresponding UVCal parameter, in parentheses. 
+For more information about the UVCal class and its parameters, please refer to pyuvdata's documentation: \url{http://pyuvdata.readthedocs.io/en/latest/uvcal.html}.
+
+A UVCal object stores calibration solutions as either a ``gain''-type solution, or as a ``delay''-type solution.
+These types represent two distinct calibration conventions, both widely used in radio astronomy.
+Depending on the calibration type (gain or delay), the \textit{calfits} format may consist of up to 4 HDUs. 
+In either case, the primary header is of the same basic format and consists of relevant meta-information for a UVCal object to be instantiated.
+It should be noted that while ``gain''-type and ``delay''-type \emph{.calfits} files share the same primary header axes, the shape of these are different in each case; for further information, refer to section \ref{primary}.
+The second HDU is also the same in either case and is the ANTENNAS HDU. 
+The third HDU, present only in ``delay''-type calibrations, stores flags which indicate which frequencies should have delays applied.
+The fourth HDU is completely optional for both types of calibrations, and stores data pertaining to the overall error of the calibration solution.
+
+\section{Primary Header}\label{primary}
+The primary HDU of a \emph{calfits} file is an image-type HDU with six axes.
+These axes, in order, represent 
+\begin{enumerate}
+	\item{Data --- varies between ``delay''- and ``gain''-type calibrations as follows:}
+	\begin{itemize}
+	\item Delay-type: this axis stores the concatenation of the two UVCal arrays [delay\_array, quality\_array], in that order.
+	These arrays both have the shape (Nants\_data, Nspws, Nfreqs, Ntimes, Njones), and are both flattened before concatenation, so the length of this axis will be $2\times$(Nants\_data + Nspws + Nfreqs + Ntimes + Njones), where these variables are those stored in the corresponding UVCal object.

```

Comment:
I don't think this is quite right. If they were flattened there would only be 1 axis, not 6 axes. I believe that the size of this axis will be 2 for delay-type and 3 or 4 for gain type -- essentially this axis is indexing into the different arrays.

Reply:
But if you look at calfits.py lines 243-274, you have: 

    # define data section based on calibration type
        `if self.cal_type == "gain":
            if self.input_flag_array is not None:
                pridata = np.concatenate(
                    [
                        self.gain_array.real[:, :, :, :, :, np.newaxis],
                        self.gain_array.imag[:, :, :, :, :, np.newaxis],
                        self.flag_array[:, :, :, :, :, np.newaxis],
                        self.input_flag_array[:, :, :, :, :, np.newaxis],
                        self.quality_array[:, :, :, :, :, np.newaxis],
                    ],
                    axis=-1,
                )
            else:
                pridata = np.concatenate(
                    [
                        self.gain_array.real[:, :, :, :, :, np.newaxis],
                        self.gain_array.imag[:, :, :, :, :, np.newaxis],
                        self.flag_array[:, :, :, :, :, np.newaxis],
                        self.quality_array[:, :, :, :, :, np.newaxis],
                    ],
                    axis=-1,
                )
        elif self.cal_type == "delay":
            pridata = np.concatenate(
                [
                    self.delay_array[:, :, :, :, :, np.newaxis],
                    self.quality_array[:, :, :, :, :, np.newaxis],
                ],
                axis=-1,
            )`

As far as I can tell, this is flattening these axes. I also thought that a single axis of a .fits file could only store a 1d-array shaped thing, but I would believe it if I'm wrong about that.

Reply:
@bhazelton you're completely right, and I had just made a silly mistake with reading list slicing wrong! I'll fix this asap. Thanks for the heads-up!

Reply:
I think my next commit should fix everything here.